Gems/Atom/Utils/Code/Include/Atom/Utils/ImGuiShaderUtils.inl

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <imgui/imgui.h>
+#include <AzCore/Serialization/Json/JsonUtils.h>
+
+namespace AZ::Render
+{
+    namespace ImGuiShaderUtils
+    {
+        inline void DrawShaderVariantTable(const AZ::RPI::ShaderOptionGroupLayout* layout, AZ::RPI::ShaderVariantId requestedVariantId, AZ::RPI::ShaderVariantId selectedVariantId)
```

Comment:
Same comment as above, can these inline functions be moved to a CPP file?

Reply:
They are in a .inl file, which is #included into the .h file. They must be inline to avoid multiple-definition compiler errors. This is the same pattern used for all ImGui related files in Atom/Utils.

Reply:
> They are in a .inl file, which is #included into the .h file. They must be inline to avoid multiple-definition compiler errors. This is the same pattern used for all ImGui related files in Atom/Utils.

This isn't a very satisfying answer. Is there an underlying reason why the current system includes the inl file in the header. Otherwise you just seem to be stating that if you remove the `inline` keyword you get multiple definition errors, which is not exactly surprising.

Reply:
I suspect the compiler will refuse to inline some of these since they are quite large.  In any case, I agree that they should probably be moved to a .cpp file.